core/libp2p/multi/converters/converter_utils.cpp

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/multi/converters/converter_utils.hpp"
+
+#include <boost/asio/ip/address_v4.hpp>
+#include <boost/algorithm/string.hpp>
+#include <outcome/outcome.hpp>
+#include "common/buffer.hpp"
+#include "common/hexutil.hpp"
+#include "libp2p/multi/converters/conversion_error.hpp"
+#include "libp2p/multi/converters/ip_v4_converter.hpp"
+#include "libp2p/multi/converters/ipfs_converter.hpp"
+#include "libp2p/multi/converters/tcp_converter.hpp"
+#include "libp2p/multi/converters/udp_converter.hpp"
+#include "libp2p/multi/multibase_codec/multibase_codec_impl.hpp"
+#include "libp2p/multi/utils/protocol_list.hpp"
+#include "libp2p/multi/utils/uvarint.hpp"
+
+using kagome::common::unhex;
+
+namespace libp2p::multi::converters {
+
+  using kagome::common::Buffer;
+
+  outcome::result<Buffer> multiaddrToBytes(std::string_view str) {
+    if (str[0] != '/') {
+      return ConversionError::ADDRESS_DOES_NOT_BEGIN_WITH_SLASH;
+    }
+    str.remove_prefix(1);
+    if(str.empty()) {
+      return ConversionError ::INVALID_ADDRESS;
+    }
+    if(str.back() == '/') {
+      str.remove_suffix(1); // for split not to recognize an empty token in the end
+    }
+
+    std::string processed;
+
+    enum class WordType { Protocol, Address };
+    WordType type = WordType::Protocol;
+
+    Protocol const *protx = nullptr;
+
+    std::list<std::string> tokens;
+    boost::algorithm::split(tokens, str, boost::algorithm::is_any_of("/"));
+
+    for (auto& word: tokens) {
+      if (type == WordType::Protocol) {
+        protx = ProtocolList::get(word);
+        if (protx != nullptr) {
+          processed += UVarint(static_cast<uint64_t>(protx->code)).toHex();
```

Comment:
Because if to simply concat bytes they would be put in buffer a bit differently, which breaks existing code, and additional processing to change this representation would hardly be much simpler than the way it is now.